org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                        Thread.Sleep(9000);
+                        MakeRequest(current_id, url, properties, attempt + 1);
+                    }
+                }
+                if ( response.StatusCode != HttpStatusCode.OK )
+                {
+                    throw new Exception(response.Content);
+                }
+            }
+            catch ( Exception e )
+            {
+                var json = $"{{\"properties\": {JsonConvert.SerializeObject(properties)} }}";
+                ExceptionLogService.LogException(new Exception($"Hubspot Sync Error{Environment.NewLine}{e}{Environment.NewLine}Current Id: {current_id}{Environment.NewLine}Exception from Request:{Environment.NewLine}{e.Message}{Environment.NewLine}Request:{Environment.NewLine}{json}{Environment.NewLine}"));
+            }
+        }
+
+        private void WriteToLog(string message)
+        {
+            string logFile = System.Web.Hosting.HostingEnvironment.MapPath("~/App_Data/Logs/HubSpotPatchLog.txt");
+            using ( System.IO.FileStream fs = new System.IO.FileStream(logFile, System.IO.FileMode.Append, System.IO.FileAccess.Write) )
+            {
+                using ( System.IO.StreamWriter sw = new System.IO.StreamWriter(fs) )
+                {
+                    sw.WriteLine(message);
+                }
+            }
+        }
+
+        private void GetContacts(string url)
+        {
+            request_count++;
+            var contactClient = new RestClient(url);
+            contactClient.Timeout = -1;
+            var contactRequest = new RestRequest(Method.GET);
+            contactRequest.AddHeader("Authorization", $"Bearer {key}");
+            IRestResponse contactResponse = contactClient.Execute(contactRequest);
+            var contactResults = JsonConvert.DeserializeObject<HSContactQueryResult>(contactResponse.Content);
+            //contacts.AddRange(contactResults.results.Where(c => c.properties.rock_person_id != null && c.properties.rock_person_id != "" && c.properties.email != null && c.properties.email != "").ToList());
+            contacts.AddRange( contactResults.results.Where( c => c.properties.rock_person_id != null && c.properties.rock_person_id != "" ).ToList() );
+            if ( contactResults.paging != null && contactResults.paging.next != null && !String.IsNullOrEmpty(contactResults.paging.next.link) && request_count < 2000 )
+            {
+                GetContacts(contactResults.paging.next.link);
+            }
+        }
+
+        private string ConvertDate(DateTime? date)
+        {
+            if ( date.HasValue )
+            {
+                DateTime today = RockDateTime.Now;
+                if ( today.Year - date.Value.Year < 1000 && today.Year - date.Value.Year > -1000 )
+                {
+                    date = new DateTime(date.Value.Year, date.Value.Month, date.Value.Day, 0, 0, 0);
+                    var d = date.Value.Subtract(new DateTime(1970, 1, 1)).TotalSeconds * 1000;
```

Comment:
The `DateTime` object has a `Date` property which is the `DateTime`'s date at midnight. These two lines could be simplified as:
```csharp
var d = date.Value.Date.Subtract(new DateTime(1970, 1, 1)).TotalSeconds * 1000;
```